x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/Sequence.java

```diff
@@ -81,7 +81,11 @@ public long ramBytesUsed() {
 
     @Override
     public int compareTo(Sequence o) {
-        return ordinal().compareTo(o.ordinal());
+        int result = ordinal().compareTo(o.ordinal());
+        if (result == 0) {
+            return this.key().compareTo(o.key());
```

Comment:
No need to qualify the variable - `key().compareTo()` - see the code style in the line above.